client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -444,7 +565,9 @@ void main(void)
 	// Note: clarity = (1 - fogginess)
 	float clarity = clamp(fogShadingParameter
 		- fogShadingParameter * length(eyeVec) / fogDistance, 0.0, 1.0);
-	col = mix(fogColor, col, clarity);
+	float fogColorMax = max(max(fogColor.r, fogColor.g), fogColor.b);
+	if (fogColorMax < 0.0000001) fogColorMax = 1.0;
```

Comment:
Use clamp instead?

Reply:
No, this would not work. The issue that this line is preventing is that a black fog color leads to a zero division, so if values of `fogColorMax` are low we should not divide by it. It may be possible to use other GLSL functions to the same effect, but the if-statement as far as I can tell is the most readable solution in this case. GLSL compilers aren't stupid either, so this probably doesn't impact performace more either.